src/main/java/com/minecolonies/rankup/util/AccountingUtils.java

```diff
@@ -0,0 +1,524 @@
+    {
+        try
+        {
+            Statement stmt = getConn().createStatement();
+
+            stmt.execute("CREATE TABLE IF NOT EXISTS " + table_id + "("
+                           + UUID_COLUMN + " varchar(255) NOT NULL, "
+                           + PLAYER_NAME_COLUMN + " varchar(255) NOT NULL, "
+                           + JOIN_DATE_COLUMN + " DATE NOT NULL, "
+                           + LAST_JOIN_COLUMN + " DATE NOT NULL, "
+                           + TIME_PLAYED_COLUMN + " int NOT NULL, "
+                           + "PRIMARY KEY(" + UUID_COLUMN + ") )");
+        }
+        catch (SQLException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    public String getPlayerName(final UUID uuid)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                final ResultSet results = getQuery("SELECT " + PLAYER_NAME_COLUMN + " FROM " + table_id + " WHERE " + UUID_COLUMN + " = '" + uuid + "'");
+
+                if (results != null && results.next())
+                {
+                    return results.getString(PLAYER_NAME_COLUMN);
+                }
+            }
+            else
+            {
+                return getAccountConfig().playerData.get(uuid).playerName;
+            }
+        }
+        catch (SQLException | NoModuleException e)
+        {
+            e.printStackTrace();
+        }
+        return "";
+    }
+
+    public String getPlayerJoinDate(final UUID uuid)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                final ResultSet results = getQuery("SELECT " + JOIN_DATE_COLUMN + " FROM " + table_id + " WHERE " + UUID_COLUMN + " = '" + uuid + "'");
+
+                if (results != null && results.next())
+                {
+                    return results.getDate(JOIN_DATE_COLUMN).toString();
+                }
+            }
+            else
+            {
+                return getAccountConfig().playerData.get(uuid).joinDate;
+            }
+        }
+        catch (SQLException | NoModuleException e)
+        {
+            e.printStackTrace();
+        }
+        return "";
+    }
+
+    public String getPlayerLastDate(final UUID uuid)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                final ResultSet results = getQuery("SELECT " + LAST_JOIN_COLUMN + " FROM " + table_id + " WHERE " + UUID_COLUMN + " = '" + uuid + "'");
+
+                if (results != null && results.next())
+                {
+                    return results.getDate(LAST_JOIN_COLUMN).toString();
+                }
+            }
+            else
+            {
+                return getAccountConfig().playerData.get(uuid).lastVisit;
+            }
+        }
+        catch (SQLException | NoModuleException e)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1166)